Plugin~/WebRTCPlugin/Codec/NvCodec/NvEncoderCuda.cpp

```diff
@@ -0,0 +1,18 @@
+﻿#include "pch.h"
+#include "nvEncodeAPI.h"
+#include "NvEncoderCuda.h"
+#include "GraphicsDevice/IGraphicsDevice.h"
+#include "GraphicsDevice/ITexture2D.h"
+
+namespace WebRTC {
+
+    NvEncoderCuda::NvEncoderCuda(const uint32_t nWidth, const uint32_t nHeight, IGraphicsDevice* device) :
+        NvEncoder(NV_ENC_DEVICE_TYPE_CUDA, NV_ENC_INPUT_RESOURCE_TYPE_CUDAARRAY, nWidth, nHeight, device)
+    {
+    }
+
+    void* NvEncoderCuda::AllocateInputResourceV(ITexture2D* tex) {
+        return tex->GetEncodeTexturePtrV();
+    }
```

Comment:
would we delete this funtion? maybe we can call tex->GetEncodeTexturePtrV() directly.

Reply:
No, we can't. The reason is because NvEncoder needs another structure if we use OpenGL.
Check NvEncoderOpenGL.cpp

Reply:
OpenGLTexture2D::GetNativeTexturePtrV return NV_ENC_INPUT_RESOURCE_OPENGL_TEX is ok?

Reply:
This is the code for OpenGL. But we are not modifying the OpenGL code in this review, and so far it has been working.

```
void* OpenGLTexture2D::GetNativeTexturePtrV() { return &m_texture; }
const void* OpenGLTexture2D::GetNativeTexturePtrV() const { return &m_texture; };
void* OpenGLTexture2D::GetEncodeTexturePtrV() { return (void*)m_texture; }
const void* OpenGLTexture2D::GetEncodeTexturePtrV() const { return (const void*)m_texture; }

```

Reply:
ok.